scripts/build-binary

```diff
@@ -25,13 +25,21 @@ VERSION_FLAGS="
         -X ${K3S_PKG}/pkg/version.Version=${VERSION}
         -X ${K3S_PKG}/pkg/version.GitCommit=${REVISION}
 "
+
+#STATIC_FLAGS='-extldflags "-static -Wl,--fatal-warnings"'
+### ON GLIBC SYSTEMS YOU WILL SEE LINKER WARNINGS #####################################################################
+# github.com/rancher/rke2
+#/usr/bin/ld: /tmp/go-link-743463534/000027.o: in function `nvmlInit_dl':
+#/home/jacob/go/pkg/mod/github.com/mindprince/gonvml@v0.0.0-20190828220739-9ebdce4bb989/bindings.go:152: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
+### THESE CAN BE SAFELY IGNORED UNLESS YOU ARE TRYING TO LEVERAGE NVIDIA METRICS FROM CADVISOR AND YOU HAPPEN TO BE ###
+### RUNNING RKE2 ON A SYSTEM OTHER THAN THE ONE WHICH IS WAS BUILT THAT ALSO HAS AN INCOMPATIBLE GLIBC ################
```

Comment:
Do we want to patch the nvml stuff out of our cadvisor fork?

Reply:
We do not see this warning when building in CI, aka using the alpine-based goboring and I imagine some folks will want nvidia metrics at some point and I would prefer to see reports of panics over reports that "I cannot get nvidia metrics out of my installation". If we want to patch the cadvisor stuff I am fine with that but it should be a separate issue.

Reply:
It looks like the cadvisor nvml stuff ends up calling `dlopen("libnvidia-ml.so.1", RTLD_LAZY);` at https://github.com/mindprince/gonvml/blob/master/bindings.go#L152.

With musl libc, dlopen is a stub call for static binaries:
* https://github.com/alandekok/jlibtool/issues/3#issue-16398345
* https://git.musl-libc.org/cgit/musl/tree/src/ldso/dlopen.c